src/arch/svm/SpokeUtils.ts

```diff
@@ -1115,15 +1121,71 @@ export const hasCCTPV1MessageBeenProcessed = async (
  * Returns the account metas for a tokenless message.
  * @returns The account metas for a tokenless message.
  */
-export async function getAccountMetasForTokenlessMessage(): Promise<IAccountMeta<string>[]> {
-  const statePda = await getStatePda(SvmSpokeClient.SVM_SPOKE_PROGRAM_ADDRESS);
-  return [
+export async function getAccountMetasForTokenlessMessage(
+  solanaClient: SVMProvider,
+  signer: KeyPairSigner,
+  messageBytes: string
+): Promise<IAccountMeta<string>[]> {
+  const deriveRootBundlePda = async (rootBundleId: number) => {
```

Comment:
You can shrink the diff here if you use [getRootBundlePda()](https://github.com/across-protocol/sdk/blob/5eb3a88283fcc4b573e94cdda0e899fd248d559b/src/arch/svm/utils.ts#L314)